src/components/views/Profile.tsx

```diff
@@ -1,17 +1,42 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import { Button } from "components/ui/Button";
 import "styles/views/Login.scss";
 import "styles/views/Profile.scss";
 import BaseContainer from "components/ui/BaseContainer";
 import { format, isValid } from "date-fns";
 import ProfileIcon from "components/ui/ProfileIcon";
+import { api, handleError } from "helpers/api";
+import User from "models/User";
+
 
 const Profile = () => {
     const navigate = useNavigate();
-    const [username, setUsername] = useState("xxGamerxx");
+    const user = localStorage.getItem("currentUser");
+    localStorage.setItem("test", user);
+    const [username, setUsername] = useState();
     const [birthday, setBirthday] = useState<string>("not given");
     const [isEditing, setIsEditing] = useState(false);
+    const [userData, setUserData] = useState({username: "...", status: "..."});
+
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const uName = localStorage.getItem("username");
+                setUsername(uName);
+                const response = await api.get("/users/" + uName);
+                const userdata = new User(response.data);
+                console.log(userdata);
+                setUserData(userdata)
+            } catch (error) {
+                alert("Server Connection Lost");
```

Comment:
write an error message that shows what the intention was. Something like the information could not be displayed because the information could not be retrieved from the server. I had no idea what was happening when I tested your code.

Reply:
and console log errors